src/lib/RoutingManager.ts

```diff
@@ -36,7 +36,7 @@ class RoutingManager implements Widget {
     this.currentUiState = this.stateMapping.routeToState(this.router.read());
   }
 
-  private getAllSearchParameters({
+  public getAllSearchParameters({
```

Comment:
I wonder if we should even have these public / private markers 🤔 since they aren't _actually_ protecting anything from consumers from using it, except if they use TS

Reply:
The presence of those markers is enforced by [@typescript-eslint/explicit-member-accessibility](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-member-accessibility.md).

I think they are useful since they are protecting us (and some consumers) to use private methods. They also could potentially make our lives easier if we migrate to native [private methods](https://github.com/tc39/proposal-private-methods) in the future.

But for consumers that do not use TS I think we should also prefix the method names with a `_` since it is common to indicate a private property/method like this in JavaScript (this could be enforced by a tool like ESLint).